### PR TITLE
@W-9648595@: Updated dependencies

### PR DESCRIPTION
**Pull request by jfeingold35**
This PR updates the following dependencies:
- `isbinaryfile`
- `@oclif/plugin-help`
- `@oclif/test`
- `@salesforce/dev-config`
- `@types/mocha`
- `@types/node`
- `mocha`
- `sinon`
- `retire`

This necessitated changes in:
- `run.ts`
- `RunOutputProcessor.ts`
- `run.js` (messages file)
- `RetireJsEngine.ts`

The following dependencies were NOT updated, although more recent major versions do exist:
- `@salesforce/command`: We use v4. While v5 is available, I identified [an issue](https://github.com/forcedotcom/cli/issues/1470) with the way tables are logged, and this version is unsuitable for us until a fix is merged and published.
- `globby`: We use v11. While v13 is available, it is ESM-based instead of CommonJS-based, and would require us to make several seemingly-risky changes. I would prefer to wait to update this dependency until after 3.0 is released.
- `tildify`: We use v2. Though v3 is available, it is ESM-based, so I skipped this for the same reason as I skipped `globby`.

### CODE REVIEW

File: package.json
Comment (jfeingold35) [inline package.json:29]: [3PP](https://gus.lightning.force.com/lightning/r/ADM_Third_Party_Software__c/a0qEE0000002Xj3YAE/view) awaiting manual approval from legal.

File: package.json
Comment (jfeingold35) [inline package.json:57]: [3PP](https://gus.lightning.force.com/lightning/r/a0qEE0000002XuLYAU/view) awaiting manual approval from legal.

File: package.json
Comment (jfeingold35) [inline package.json:58]: [3PP](https://gus.lightning.force.com/lightning/r/ADM_Third_Party_Software__c/a0qEE0000002XmHYAU/view) awaiting manual approval from legal.

File: package.json
Comment (jfeingold35) [inline package.json:59]: No 3PP logged, since we own this package.

File: package.json
Comment (jfeingold35) [inline package.json:66]: [3PP](https://gus.lightning.force.com/lightning/r/ADM_Third_Party_Software__c/a0qEE0000002XntYAE/view) awaiting manual pproval from legal.

File: package.json
Comment (jfeingold35) [inline package.json:68]: [3PP](https://gus.lightning.force.com/lightning/r/ADM_Third_Party_Software__c/a0qEE0000002XpVYAU/view) awaiting manual approval from legal.

File: package.json
Comment (jfeingold35) [inline package.json:76]: [3PP](https://gus.lightning.force.com/lightning/r/ADM_Third_Party_Software__c/a0qEE0000002Xr7YAE/view) awaiting manual security review.

File: package.json
Comment (jfeingold35) [inline package.json:79]: [3PP](https://gus.lightning.force.com/lightning/r/ADM_Third_Party_Software__c/a0qEE0000002XsjYAE/view) approved.

File: messages/run.js
Comment (jfeingold35) [inline messages/run.js:37]: This message was out of date.

File: src/commands/scanner/run.ts
Comment (jfeingold35) [inline src/commands/scanner/run.ts:208]: Previously, if you tried to specify `--format table --outfile whatever`, it would throw the out-of-date message about not having an acceptable file extension. Now, we explicitly throw an exception indicating that `table` can't be written to a file. This makes sense, because the table format is intended to be human-readable.

File: src/commands/scanner/run.ts
Comment (jfeingold35) [inline src/commands/scanner/run.ts:229]: Renamed this method, because "infer" felt more technically correct to me.

File: src/lib/util/RunOutputProcessor.ts
Comment (jfeingold35) [inline src/lib/util/RunOutputProcessor.ts:118]: Updating `@types/node` to v17 caused this to break. The validation added in `run.ts` allows us to safely assume that `results` is a string, so we can cast without issue.

**Review by rmohan20 — APPROVED** [review]
Thanks for the thorough work. Looks great! I had a question about RetireJS version that I posted on our team channel. Please let me know your thoughts. In the meantime, approving this PR to unblock you.

File: package.json
Comment (jfeingold35) [inline package.json:35]: [3PP](https://gus.lightning.force.com/lightning/r/ADM_Third_Party_Software__c/a0qEE0000002YgjYAE/view) pending approval.

File: src/lib/retire-js/RetireJsEngine.ts
Comment (jfeingold35) [inline src/lib/retire-js/RetireJsEngine.ts:224]: As of [retire v3.0.2](https://github.com/RetireJS/retire.js/blob/master/node/CHANGELOG.md#302), the JSON is always written to stdout.

**Pull request merged**
